flink-connector/flink/build.gradle.kts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+  id("idea")
+}
+
+repositories {
+  mavenCentral()
+}
+
+val flinkVersion: String = libs.versions.flink.get()
+
+// The Flink only support scala 2.12, and all scala api will be removed in a future version.
+// You can find more detail at the following issues:
+// https://issues.apache.org/jira/browse/FLINK-23986,
+// https://issues.apache.org/jira/browse/FLINK-20845,
+// https://issues.apache.org/jira/browse/FLINK-13414.
+val scalaVersion: String = "2.12"
+val artifactName = "gravitino-${project.name}_$scalaVersion"
+
+dependencies {
+  implementation(project(":api"))
+  implementation(project(":catalogs:catalog-common"))
+  implementation(project(":common"))
+  implementation(project(":core"))
+
+  implementation(libs.bundles.log4j)
+  implementation(libs.commons.lang3)
+  implementation(libs.guava)
+  implementation(libs.httpclient5)
+  implementation(libs.jackson.databind)
+  implementation(libs.jackson.annotations)
+  implementation(libs.jackson.datatype.jdk8)
+  implementation(libs.jackson.datatype.jsr310)
+
+  implementation("org.apache.flink:flink-connector-hive_$scalaVersion:$flinkVersion")
```

Comment:
should we package flink jar to the flink connector? as the flink connector is run under the flink enviroment.

Reply:
@FANNG1 I'm not sure if we should package the flink-connector-hive. Should we assume that the engine side already has it？Flink does not add this dependency by default and users need to download it themselves. https://nightlies.apache.org/flink/flink-docs-release-1.17/zh/docs/connectors/table/hive/overview/#%e4%bd%bf%e7%94%a8-flink-%e6%8f%90%e4%be%9b%e7%9a%84-hive-jar
WDYT?

Reply:
I perfer not to add the jar because most users using have should download the jar and may conflict with the `flink-connector-hive`,  same for `flink-table-common` `flink-table-api-java`